### PR TITLE
NATS Object Store in the Go runner

### PR DESCRIPTION
**Pull request by dfinteligenz**
## Description

New Object Store object added to context.
Moved errors to new internal errors package.
Now runner and context use structures as params.

## Motivation and Context

As a user I want to use the NATS Object Store in my nodes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other changes (ci configuration, documentation or any other kind of changes)

## Checklist:

- [ ] I have created tests for my code changes, and the tests are passing.
- [ ] I have executed the pre-commit hooks locally.
- [x] My change requires a change to the documentation (create a new issue if the documentation has not been updated).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by kafkaphoenix — DISMISSED** [review]
Good job

**Review by kafkaphoenix — APPROVED** [review]
Good job!

File: kre-go/context.go [hunk 1]
```diff
@@ -126,26 +146,44 @@ func (c *HandlerContext) SendOutput(response proto.Message, channelOpt ...string
...
 
 // SendEarlyExit works as the SendOutput functionality
 // with the addition of typing this message as an early exit.
 func (c *HandlerContext) SendEarlyExit(response proto.Message, channelOpt ...string) error {
-	return c.publishMsg(response, c.reqMsg, MessageType_EARLY_EXIT, c.getChannel(channelOpt))
+	return c.publishMsg(response, c.reqMsg, MessageType_EARLY_EXIT, c.getOptionalString(channelOpt))
+}
+
+// StoreObject stores the given payload in the Object Store with the given key as identifier
+// If an objectStore name is given, a new Object Store will be created.
```
Comment (JYisus) [inline kre-go/context.go:165]: Outdated

File: kre-go/context.go [hunk 2]
```diff
@@ -126,26 +146,44 @@ func (c *HandlerContext) SendOutput(response proto.Message, channelOpt ...string
...
+func (c *HandlerContext) StoreObject(key string, payload []byte) error {
+	return c.storeObject(key, payload)
+}
+
+// GetObject retrieves the object stored in the given object store or the default object store for the workflow
```
Comment (JYisus) [inline kre-go/context.go:170]:
```suggestion
// GetObject retrieves the object stored in the node's object store
```

File: kre-go/context.go [hunk 3]
```diff
@@ -126,26 +146,44 @@ func (c *HandlerContext) SendOutput(response proto.Message, channelOpt ...string
...
+// with the given key as identifier as a byte array.
+func (c *HandlerContext) GetObject(key string) ([]byte, error) {
+	return c.getObject(key)
+}
+
+// DeleteObject deletes the object stored in the given object store or the default object store for the workflow
```
Comment (JYisus) [inline kre-go/context.go:176]:
```suggestion
// DeleteObject deletes the object stored in the node's object store
```

File: kre-go/context_objectStore.go
Comment (dfinteligenz) [inline kre-go/context_objectStore.go:1]: I'd rename this file to all lowercase.

`context_objectstore.go`

Comment (kafkaphoenix) [reply kre-go/context_objectStore.go:1]: yes i agree with that it looks terrible otherwise

**Review by kafkaphoenix — APPROVED** [review]

**Review by alexrodfe — DISMISSED** [review]

**Review by JYisus — APPROVED** [review]

**Review by alexrodfe — APPROVED** [review]
great job!

**Pull request merged**